packages/material-ui/src/BusyButton/BusyButton.d.ts

```diff
@@ -0,0 +1,47 @@
+import { ExtendButton, ExtendButtonTypeMap } from '../Button';
+import { OverrideProps } from '../OverridableComponent';
+
+export type BusyButtonTypeMap<
+  P = {},
+  D extends React.ElementType = 'button'
+> = ExtendButtonTypeMap<{
+  props: P & {
+    /**
+     * The content of the button.
+     */
+    pending?: boolean;
+    /**
+     * Element placed before the children if the button is in pending state.
+     */
+    pendingIndicator?: React.ReactNode;
+    /**
+     * The pending indicator can be positioned on the start, end or the center of the Button.
+     */
+    pendingIndicatorPosition?: 'start' | 'end' | 'center';
```

Comment:
While I imagine developers would set the prop once, in their default theme prop configuration, I think that we could shorten the name of the prop without losing much.

```suggestion
    pendingPosition?: 'start' | 'end' | 'center';
```